File: analysis/analysis.R
# analysis.R — Trastornos de Salud Mental en Población Universitaria Latinoamericana: Prevalencia, Factores de Riesgo y Variables Protectoras — Base de Datos Estadística
# DOI: 10.5281/zenodo.18815856
# Author: de la Serna Tuya, Juan Moisés · ORCID: 0000-0002-8401-8018
# License: CC0 1.0

# ── SETUP ──────────────────────────────────────────────────────────────────
# Install required packages if needed:
# install.packages(c("readr","ggplot2","dplyr","tidyr"))

library(readr)
library(ggplot2)
library(dplyr)

# ── LOAD DATA ──────────────────────────────────────────────────────────────
# Download dataset from: https://doi.org/10.5281/zenodo.18815856
# df <- read_csv("dataset.csv")

# Example with synthetic data
set.seed(42)
years <- 2000:2023
df <- data.frame(
  year = years,
    salud_mental = rnorm(length(years), 50, 15),
  universitari = rnorm(length(years), 50, 15),
  américa_lati = rnorm(length(years), 50, 15),
  depresión = rnorm(length(years), 50, 15),
  stringsAsFactors = FALSE
)

cat("Dataset: Trastornos de Salud Mental en Población Universita\n")
cat("DOI: 10.5281/zenodo.18815856\n")
cat("Dimensions:", nrow(df), "x", ncol(df), "\n\n")
print(head(df))
print(summary(df))

# ── VISUALIZATION ──────────────────────────────────────────────────────────
# Temporal trend
p1 <- df %>%
  tidyr::pivot_longer(-year, names_to="variable", values_to="value") %>%
  ggplot(aes(x=year, y=value, color=variable)) +
  geom_line(size=0.8) +
  geom_point(size=1.5) +
  labs(title="Trastornos de Salud Mental en Población Universitaria L",
       subtitle="DOI: 10.5281/zenodo.18815856",
       x="Year", y="Value",
       caption="de la Serna Tuya, Juan Moisés · 2026") +
  theme_minimal() +
  theme(legend.position="bottom")

ggsave("figures/analysis_output_r.png", p1, width=10, height=6, dpi=150)
cat("\nFigure saved: figures/analysis_output_r.png\n")

# ── CITATION ───────────────────────────────────────────────────────────────
cat("\nCitation:\n")
cat("de la Serna Tuya, Juan Moisés (2026). Trastornos de Salud Mental en Población Universitaria Latinoamericana:. Zenodo. https://doi.org/10.5281/zenodo.18815856\n")

File: analysis/analysis.py
# analysis.py — Trastornos de Salud Mental en Población Universitaria Latinoamericana: Prevalencia, Factores de Riesgo y Variables Protectoras — Base de Datos Estadística
# DOI: 10.5281/zenodo.18815856
# Author: de la Serna Tuya, Juan Moisés · ORCID: 0000-0002-8401-8018
# License: CC0 1.0

import pandas as pd
import matplotlib.pyplot as plt
import matplotlib
matplotlib.use("Agg")
import warnings
warnings.filterwarnings("ignore")

# ── LOAD DATA ─────────────────────────────────────────────────────────────
# Download dataset from: https://doi.org/10.5281/zenodo.18815856
# df = pd.read_csv("dataset.csv")

# Example with synthetic data
import numpy as np
np.random.seed(42)
years = list(range(2000, 2024))
keywords = ['salud mental', 'universitarios', 'América Latina', 'depresión', 'ansiedad']

df = pd.DataFrame({
    "year": years,
    **{k.lower().replace(" ","_")[:15]: np.random.normal(50, 15, len(years))
       for k in keywords[:4]}
})

print(f"Dataset: Trastornos de Salud Mental en Población Universita")
print(f"DOI: 10.5281/zenodo.18815856")
print(f"Shape: {df.shape}")
print("\nFirst rows:")
print(df.head())
print("\nSummary statistics:")
print(df.describe())

# ── VISUALIZATION ─────────────────────────────────────────────────────────
fig, axes = plt.subplots(1, 2, figsize=(12, 5))
fig.suptitle("Trastornos de Salud Mental en Población Universitaria Latino", fontsize=11, fontweight="bold")

# Plot 1: Temporal trend
ax1 = axes[0]
for col in df.columns[1:3]:
    ax1.plot(df["year"], df[col], marker="o", markersize=3, label=col)
ax1.set_xlabel("Year")
ax1.set_ylabel("Value")
ax1.set_title("Temporal Trends")
ax1.legend(fontsize=8)
ax1.grid(alpha=0.3)

# Plot 2: Distribution
ax2 = axes[1]
df.iloc[:, 1:5].mean().plot(kind="bar", ax=ax2, color=["#1f6feb","#f85149","#3fb950","#e3b341"])
ax2.set_title("Variable Means")
ax2.set_xlabel("Variable")
ax2.set_ylabel("Mean Value")
ax2.tick_params(axis="x", rotation=30)
ax2.grid(axis="y", alpha=0.3)

plt.tight_layout()
plt.savefig("figures/analysis_output.png", dpi=150, bbox_inches="tight")
print("\nFigure saved: figures/analysis_output.png")

# ── CITATION ──────────────────────────────────────────────────────────────
print(f"""
Citation:
de la Serna Tuya, Juan Moisés (2026). Trastornos de Salud Mental en Población Universitaria Latinoamericana:. Zenodo. https://doi.org/10.5281/zenodo.18815856
""")
